internal/transport/http2_client.go

```diff
@@ -681,13 +681,15 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (_ *Strea
 	}
 	if t.statsHandler != nil {
 		header, _, _ := metadata.FromOutgoingContextRaw(ctx)
```

Comment:
Actually, I believe this is a bug to call `Raw` here.
This should be just `FromOutgoingContext()`, otherwise it will miss the metadata that's appended.
Can you fix it, too? Thanks!

And I believe if you change this, the `copy()` won't be necessary anymore, because `FromOutgoingContext` always does a [join](https://github.com/grpc/grpc-go/blob/b88d2d74656f2b9833cc78c088ebf899d2189230/metadata/metadata.go#L203). (The comment of the function also needs update, but that's another issue).

Reply:
Done.